nemo_text_processing/g2p/modules.py

```diff
@@ -315,40 +316,29 @@ def __init__(
         else:
             self.use_chars = use_chars
 
+        # parse the dictionary, update it, and generate symbol set.
         if isinstance(phoneme_dict, str) or isinstance(phoneme_dict, pathlib.Path):
-            # Load phoneme_dict from file path
-            self.phoneme_dict, self.symbols = self._parse_as_cmu_dict(phoneme_dict)
+            # load the dictionary file where there may exist a digit suffix after a word, which
+            # represents the pronunciation variant of that word.
+            phoneme_dict_obj = defaultdict(list)
+            _alt_re = re.compile(r'\([0-9]+\)')
+            with open(phoneme_dict, "r") as fdict:
+                for line in fdict:
+                    if len(line) and ('A' <= line[0] <= 'Z' or line[0] == "'"):
+                        parts = line.strip().split()
+                        assert len(parts) == 2, f"Wrong format for the entry: {line.strip()}."
```

Comment:
I think this is equivalent to the previous implementation (which splits on the double space between an entry and its pronunciation) for the CMUdicts we have, but splitting on any whitespace is looser and that worries me a bit. We probably won't encounter an entry with a space in it, but is there any other case in which this might break something?

Should we keep it as a split on double space?

Reply:
common entries in a dictionary file usually look like the below. I was thinking to support any formats below (trying to keep away from CMUlike dict constaints) so I applied `.split()`, but added an assertion to make sure an equal functioning as the previous implementation. As shown in `_parse_as_cmu_dict`, new implementation supports any below formats already. The assertion actually can be removed. If keeping splitting on double space, then `m.split("  ")` becomes `['word_4', 'b c d']`. 
```
x = "word_1  bcd"
y = "word_2 bcd"
z = "word_3\tbcd"
m = "word_4  b c d"
n = "word_5\tb c d"
```

Reply:
I just remembered that the double-space requirement was a holdover from the ARPABET-based dictionary, where the phonemes were separated by single spaces. IPA is all mashed together so I think this should be fine (and it'll be good to support other formats), but I do like having the assertion there just to catch any accidental user error.

Reply:
ok then, I will match the previous implementation with double whitespace for this PR and will make another PR to support the flexible entry formats above. 

again, for language support expansion purposes, it would be great to make the dict format flexible.

Reply:
Wait, before you change it back, I'm agreeing--please leave as-is. It should be fine with the assertion.